src/extension/providers/liquid/index.ts

```diff
@@ -0,0 +1,40 @@
+import { postMessage } from "../postMessage";
+
+declare global {
+  interface Window {
+    liquid: LiquidProvider;
+  }
+}
+
+export default class LiquidProvider {
+  enabled: boolean;
+
+  constructor() {
+    this.enabled = false;
+  }
+
+  async enable() {
+    if (this.enabled) {
+      return { enabled: true };
+    }
+    return await this.execute("enable");
+  }
+
+  async getAddress() {
+    await this.enable();
+    return await this.execute("getAddressOrPrompt");
+  }
+
+  async signPset(pset: string) {
```

Comment:
@im-adithya I cannot see signSchnorr here. Do we want to expose it? otherwise should we remove the confirmation dialogs?

Reply:
+1 to do not expose signSchnorr not needed thanks to signPset and could be dangerous

Reply:
Thanks @louisinger for confirmation. I will remove it along with the confirmation dialogs in the other PR.

Reply:
@louisinger @im-adithya removed in https://github.com/getAlby/lightning-browser-extension/pull/2583